ci.nix

```diff
@@ -1,48 +1,74 @@
-# 'supportedSystems' restricts the set of systems that we will evaluate for. Useful when you're evaluting
-# on a machine with e.g. no way to build the Darwin IFDs you need!
-{ supportedSystems ? [ "x86_64-linux" "x86_64-darwin" ]
+{
+  # 'supportedSystems' restricts the set of systems that we will evaluate for. Useful when you're evaluting
+  # on a machine with e.g. no way to build the Darwin IFDs you need!
+  supportedSystems ? [ "x86_64-linux" "x86_64-darwin" ]
   # This will be used by the packages that get the git revision in lieu of actually trying to find it,
   # which doesn't work in all situations. Set to null to get it from git.
 , rev ? "fake"
 }:
+
 let
-  inherit (import ./nix/lib/ci.nix) dimension platformFilterGeneric filterAttrsOnlyRecursive;
-  systems = nixpkgs: nixpkgs.lib.filterAttrs (_: v: builtins.elem v supportedSystems) {
-    # I wanted to take these from 'lib.systems.examples', but apparently there isn't one for linux!
-    linux = "x86_64-linux";
-    darwin = "x86_64-darwin";
-  };
+  inherit (import ./nix/lib/ci.nix) dimension platformFilterGeneric filterAttrsOnlyRecursive filterSystems;
   sources = import ./nix/sources.nix;
-  # Useful for generic library functions: do not use for anything platform dependent
-  genericPkgs = import sources.nixpkgs { };
-in
-dimension "System" (systems genericPkgs) (systemName: system:
-  let
-    packages = import ./default.nix { inherit system rev; checkMaterialization = true; };
-    pkgs = packages.pkgs;
-    plutus = packages.plutus;
-    plutusMusl = packages.plutusMusl;
-    lib = pkgs.lib;
-    platformFilter = platformFilterGeneric pkgs system;
-  in
-  filterAttrsOnlyRecursive (_: v: platformFilter v) {
-    inherit (packages) docs tests plutus-playground marlowe-playground plutus-scb marlowe-symbolic-lambda;
-    inherit (plutus.haskell.project) roots;
+  # limit supportedSystems to what the CI can actually build
+  # currently that is linux and darwin.
+  systems = filterSystems supportedSystems;
+
+  # Collects haskell derivations and builds an attrset:
+  #
+  # { library = { ... }
+  # , tests = { ... }
+  # , benchmarks = { ... }
+  # , exes = { ... }
+  # , checks = { ... }
+  # }
+  #  Where each attribute contains an attribute set
+  #  with all haskell components of that type
+  mkHaskellDimension = pkgs: haskellProjects:
+    let
+      # retrieve all checks from a Haskell package
+      collectChecks = _: ps: pkgs.recurseIntoAttrs (builtins.mapAttrs (_: p: p.checks) ps);
```

Comment:
I actually upstreamed this into `haskell.nix`, we can probably use `collectChecks'` from there now.